module/scheduler_components/remaster_queue_manager.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <list>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "common/configuration.h"
+#include "common/constants.h"
+#include "common/types.h"
+#include "common/transaction_utils.h"
+
+#include "storage/storage.h"
+
+using std::list;
+using std::shared_ptr;
+using std::pair;
+using std::unordered_map;
+using std::unordered_set;
+using std::vector;
+
+namespace slog {
+
+enum class VerifyMasterResult {VALID, WAITING, ABORT};
+
+
+/**
+ * The remaster queue manager also conducts the check of master metadata.
+ * If a remaster has occured since the transaction was forwarded, it may
+ * need to be restarted. If the transaction arrived before a remaster that
+ * the forwarder included in the metadata, then it will need to wait.
+ */
+class RemasterQueueManager {
```

Comment:
`RemasterManager` might be a better name since it does not make the assumption for what we use to implement it.

Reply:
That makes sense, is there a better way to make it clear that this handles checking the counters rather than executing remaster transactions?

Reply:
maybe `RemasterCoordinator`?